snap/device.go

```diff
@@ -33,6 +33,10 @@ type Device interface {
 	Gadget() string
 
 	HasModeenv() bool
+	IsCoreLegacy() bool     // true if UC16/18
```

Comment:
this one seems never to be used directly, I would try to avoid needing it unless you think we need it later?

Reply:
OTOH I would prefer to keep the surface of Device as small as possible and add this only when we have a couple clear use cases which the branch doesn't have atm

Reply:
I've removed it now